src/app/infra/client-storage/local-storage.spec.ts

```diff
@@ -0,0 +1,76 @@
+import { TestBed } from '@angular/core/testing';
+import { LocalStorageService } from './local-storage.service';
+
+describe('LocalStorageService', () => {
+  let localStorageService: LocalStorageService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [LocalStorageService],
+    });
+
+    localStorageService = TestBed.inject(LocalStorageService);
+  });
+
+  afterEach(() => {
+    localStorage.clear();
+  });
+
+  describe('Set', () => {
+    it('should add value in local storage', () => {
+      localStorageService.set('folklore', 'evermore');
+      const value = JSON.parse(localStorage.getItem('folklore') || '');
+      expect(value).toBe('evermore');
+    });
+    it('should throw error when no key is provided to set', () => {
+      expect(() => {
+        localStorageService.set('', 'evermore');
+      }).toThrowError('Error saving to localStorage: key not provided');
+    });
+    it.todo(
+      'should throw error when local storage cannot set the provided value'
+    );
```

Comment:
I don't know how to test this. 😅

Reply:
> I don't know how to test this. 😅

You can use toThrow in expect.
```js
test("Test description", () => {
  const t = () => {
    throw new TypeError("UNKNOWN ERROR");
  };
  expect(t).toThrow(TypeError);
  expect(t).toThrow("UNKNOWN ERROR");
});
```

Reply:
The problem is how I can make `localStorage.setItem(key, JSON.stringify(data))` return an error.
I read that this function returns an error when user does not allow local storage to save things but I don't know how to replicate that in test environment.